src/uu/lslocks/src/lslocks.rs

```diff
@@ -0,0 +1,306 @@
+}
+
+impl Lock {
+    fn get_value(&self, col: &Column) -> String {
+        match col {
+            Column::Command => resolve_command(self).unwrap_or("<unknown>".to_string()),
+            Column::Pid => self
+                .pid
+                .map(|pid| pid.to_string())
+                .unwrap_or("-".to_string()),
+            Column::Type => self.lock_type.to_string(),
+            Column::Size => todo!(),
+            Column::Inode => self.inode.to_string(),
+            Column::MajMin => self.major_minor.clone(),
+            Column::Mode => self.mode.to_string(),
+            Column::Mandatory => {
+                if self.mandatory {
+                    "1".to_string()
+                } else {
+                    "0".to_string()
+                }
+            }
+            Column::Start => self.start_offset.to_string(),
+            // TODO: In the case of EOF end_offset, we should actually resolve the actual file size and display that as the end offset
+            Column::End => self
+                .end_offset
+                .map(|offset| offset.to_string())
+                .unwrap_or("EOF".to_string()),
+            Column::Path => todo!(), // TODO: Resolve filepath of the lock target
+            Column::Blocker => todo!(), // TODO: Check if lock is blocker (and by what)
+            Column::Holders => todo!(), // TODO: Resolve all holders of the lock (this would also let us display a process/PID for OFD locks)
+        }
+    }
+}
+
+impl FromStr for Lock {
+    type Err = ();
+    fn from_str(input: &str) -> Result<Self, Self::Err> {
+        let mut parts = input.split_whitespace();
+
+        // Ordinal position comes in the form of `<value>:`, so we need to strip away the `:`
+        let ord = parts
+            .next()
+            .and_then(|s| s.strip_suffix(":"))
+            .unwrap()
+            .parse::<usize>()
+            .unwrap();
+        let lock_type = parts
+            .next()
+            .and_then(|part| LockType::from_str(part).ok())
+            .unwrap();
+        let mandatory = parts
+            .next()
+            .map(|part| match part {
+                "MANDATORY" => true,
+                "ADVISORY" => false,
+                _ => panic!("Unrecognized value in lock line: {}", part),
+            })
+            .unwrap();
+        let mode = parts
+            .next()
+            .and_then(|part| LockMode::from_str(part).ok())
+            .unwrap();
+        let pid: Option<usize> = parts.next().and_then(|pid_str| match pid_str {
+            "-1" => None,
+            other => other.parse::<usize>().ok(),
+        });
+
+        if lock_type == LockType::OFDLCK && pid.is_some() {
+            println!("Unexpected PID value on OFD lock: '{}'", input);
+            return Err(());
+        };
+
+        // This field has a format of MAJOR:MINOR:INODE
+        let maj_min_inode: Vec<_> = parts.next().unwrap().split(":").collect();
```

Comment:
For consistency reasons I would rename this var to `major_min_inode` because on line 99 you use `major_min` as var name.